sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -293,9 +294,15 @@ class DataFrame protected[sql](
    *   1983  03    0.410516        0.442194
    *   1984  04    0.450090        0.483521
    * }}}
+   * @param numRows Number of rows to show
    * @group basic
    */
-  def show(): Unit = println(showString())
+  def show(numRows: Int = 20): Unit = println(showString(numRows))
+
+  /**
+   * Displays the [[DataFrame]] in a tabular form. (For Java compatibility)
+   */
+  def show(): Unit = println(showString(20))
```

Comment:
i.e don't include "for java compatibility" in the user facing doc, and don't include the default param value.